lib/config/flat-config-array.js

```diff
@@ -192,17 +192,7 @@ class FlatConfigArray extends ConfigArray {
         if (languageOptions && languageOptions.parser) {
             const { parser } = languageOptions;
 
-            if (typeof parser === "string") {
```

Comment:
@fasttime another nice catch. 👍 This was never being called because the schema errors before it ever gets here.